sdk/identity/azure-identity/test/ut/managed_identity_credential_test.cpp

```diff
@@ -1142,6 +1214,246 @@ TEST(ManagedIdentityCredential, AzureArcAuthHeaderTwoEquals)
       }));
 }
 
+TEST(ManagedIdentityCredential, AzureArcInvalidKey)
+{
+  using Azure::Core::Credentials::AccessToken;
+  using Azure::Core::Credentials::AuthenticationException;
+
+  std::string validKeyPath;
+
+#if defined(AZ_PLATFORM_LINUX)
+  validKeyPath = "/var/opt/azcmagent/tokens/";
+#elif defined(AZ_PLATFORM_WINDOWS)
+  validKeyPath = Azure::Core::_internal::Environment::GetVariable("ProgramData");
+  if (validKeyPath.empty())
+  {
+    GTEST_LOG_(ERROR) << "We can't get ProgramData folder path in AzureArcInvalidKey test.";
+    EXPECT_TRUE(false);
+  }
+  validKeyPath += "\\AzureConnectedMachineAgent\\Tokens\\";
+#else
+  // Unsupported OS
+  static_cast<void>(CredentialTestHelper::SimulateTokenRequest(
```

Comment:
Is this codepath executed ever? Or is this feature simply not supported on Mac?

Reply:
So instead of saying that this is for "unsupported OS", it might be nice to say that it's specific to OSX.

Reply:
The test code path is executed, the comment is about the unsupported OS, in the source. I can change it to OSX if you like.